File: proyecto/happypet/src/main/java/happypet/entities/Producto.java
package happypet.entities;

import java.util.Collection;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Builder
@AllArgsConstructor
@Entity
public class Producto {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column(unique = true)
	private String nombre;

	private String descripcion;
	
	private String precio;
	


	@OneToMany(mappedBy = "producto")
	private Collection<Comentario> comentarios;

	@JsonIgnore
	@OneToMany(mappedBy = "product")
	private Collection<Ldv> ldvs;
	
	@ManyToOne
	private Categoria categoria;
}

File: proyecto/cliente/src/app/producto/producto.ts
import { Categoria } from "../categoria/categoria";
import { Comentario } from "../comentario/comentario";
import { Ldv } from "../ldv/ldv";

export interface Producto {
    id:number;
    nombre:string;
    descripcion:string;
    precio:string;
    categoria: Categoria;
    comentarios:Comentario[];
    ldv:Ldv[];
}
